src/main/java/model/item/Equipment.java

```diff
@@ -1,50 +1,56 @@
 package model.item;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.ReadOnlyIntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
 import manager.Database;
 import manager.TypTranslation;
+import manager.Utility;
 import model.Currency;
 import model.upgrade.Upgrade;
 import model.upgrade.UpgradeFactory;
 
 import java.util.*;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 public abstract class Equipment extends Item {
 
     protected static Random rand = new Random();
 
-    protected String material = "";
-    protected int slots = 0;
-    protected ArrayList<Upgrade> upgrades = new ArrayList<>();
-
-    public String getMaterial() {
-        return material;
-    }
-
-    public void setMaterial(String material) {
-        this.material = material;
-    }
-
-    public int getSlots() {
-        return slots;
-    }
-
-    public void setSlots(int slots) {
-        this.slots = slots;
-    }
-
-    public Collection<Upgrade> getUpgrades() {
-        return upgrades;
+    protected String material;
```

Comment:
This should not be a string and you know it.. Material could be a class with tier at least!

Reply:
Yes, that would require a seperate table in the database.
I would say that is an own feature.

Reply:
please open a feature request issue